xmtp_cursor_state/src/store.rs

```diff
@@ -0,0 +1,227 @@
+use std::collections::HashMap;
+use std::fmt;
+use std::sync::{Arc, Mutex};
+use xmtp_proto::xmtp::xmtpv4::envelopes::Cursor;
+
+pub type SharedCursorStore = Arc<Mutex<CursorStore>>;
+
+pub struct CursorStore {
+    topics: HashMap<Vec<u8>, Cursor>,
+}
+
+impl CursorStore {
+    pub fn new() -> Self {
+        Self {
+            topics: HashMap::new(),
+        }
+    }
+
+    /// Record that a message for this topic with the given clock was processed
+    pub fn processed(&mut self, topic: Vec<u8>, new_clock: &Cursor) {
```

Comment:
I think we should make a distinction here between 'processed' and 'received' -- processed is maybe overused within the mls side of things. But for my own understanding, this isn't doing anything with the message, but its just recording that we received a message with the given clock (is that right?)